### DISCUSSION TITLE
Is there a way to expose number of tokens used in a run or in a graph?

### DISCUSSION THREAD

**Discussion by trancethehuman** [ScrapeGraphAI/Scrapegraph-ai#252] [Q&A]
Need to keep track of my costs :)

**Comment by f-aguzzi** [comment]
[`scrapegraphai/utils/prettify_exec_info.py`](https://github.com/VinciGit00/Scrapegraph-ai/blob/main/scrapegraphai/utils/prettify_exec_info.py) should do the job. I don't remember if it's referenced by other classes or if you have to run it yourself, so this is an incomplete answer, but I hope it can give you a pointer to the right direction.

**Comment by VinciGit00** [comment]
Go to smart scraper OpenAI, there should be the function

**Comment by VinciGit00** [comment]
also mistral ow have it
